code/modules/mapping/reader.dm

```diff
@@ -163,31 +248,193 @@
 	. = _load_impl(x_offset, y_offset, z_offset, cropMap, no_changeturf, x_lower, x_upper, y_lower, y_upper, placeOnTop)
 	Master.StopLoadingMap()
 
-
+/*
 #define MAPLOADING_CHECK_TICK \
 	if(TICK_CHECK) { \
 		SSatoms.map_loader_stop(); \
 		stoplag(); \
 		SSatoms.map_loader_begin(); \
 	}
+*/
+#define MAPLOADING_CHECK_TICK
+
 // Do not call except via load() above.
 /datum/parsed_map/proc/_load_impl(x_offset = 1, y_offset = 1, z_offset = world.maxz + 1, cropMap = FALSE, no_changeturf = FALSE, x_lower = -INFINITY, x_upper = INFINITY, y_lower = -INFINITY, y_upper = INFINITY, placeOnTop = FALSE)
 	PRIVATE_PROC(TRUE)
+	// Tell ss atoms that we're doing maploading
+	// We'll have to account for this in the following tick_checks so it doesn't overflow
+	SSatoms.map_loader_begin()
+
+	// Loading used to be done in this proc
+	// We make the assumption that if the inner procs runtime, we WANT to do cleanup on them, but we should stil tell our parents we failed
+	// Since well, we did
+	var/sucessful = FALSE
+	switch(map_format)
+		if(MAP_TGM)
+			sucessful = _tgm_load(x_offset, y_offset, z_offset, cropMap, no_changeturf, x_lower, x_upper, y_lower, y_upper, placeOnTop)
+		else
+			sucessful = _dmm_load(x_offset, y_offset, z_offset, cropMap, no_changeturf, x_lower, x_upper, y_lower, y_upper, placeOnTop)
+
+	// And we are done lads, call it off
+	SSatoms.map_loader_stop()
+
+	if(!no_changeturf)
+		for(var/turf/T as anything in block(locate(bounds[MAP_MINX], bounds[MAP_MINY], bounds[MAP_MINZ]), locate(bounds[MAP_MAXX], bounds[MAP_MAXY], bounds[MAP_MAXZ])))
```

Comment:
Better name

Also are we sure `as anything in block` is faster than `in block` for turfs? I know we have an answer for this somewhere

Reply:
`as anything` should be on par? as `as turf` as of 515, because `block` inherited the internal opts that `view()` previously had with the `as` keyword.

Reply:
there should be no reason for it to be slower, and this way I don't need to worry about if byond is smart or not